packages/core/src/models/Tools.ts

```diff
@@ -48,7 +48,7 @@ import { removeNonRequiredProperties } from '../utils/jsonSchema';
 export class Tools<
   TToolCollection,
   TTool,
-  TProvider extends BaseComposioProvider<TToolCollection, TTool>,
+  TProvider extends BaseComposioProvider<TToolCollection, TTool, unknown>,
```

Comment:
We shouldn't be using `unkown` here, if we are using it then there is something wrong with the definition of BaseComposioProvider that needs to be fixed.